pkg/kubectl/describe.go

```diff
@@ -878,10 +878,33 @@ func describeNode(node *api.Node, pods []*api.Pod, events *api.EventList) (strin
 			fmt.Fprintf(out, "ExternalID:\t%s\n", node.Spec.ExternalID)
 		}
 		fmt.Fprintf(out, "Pods:\t(%d in total)\n", len(pods))
-		fmt.Fprint(out, "  Namespace\tName\n")
+		fmt.Fprint(out, "  Namespace\tName\t\tCPU(milliCPU)\t\tMemory(bytes)\n")
+		totalMilliCPU := int64(0)
+		totalMemory := int64(0)
+		fractionPodCPU := float64(0)
+		fractionPodMemory := float64(0)
+		fractionTotalCPU := float64(0)
+		fractionTotalMemory := float64(0)
 		for _, pod := range pods {
-			fmt.Fprintf(out, "  %s\t%s\n", pod.Namespace, pod.Name)
+			podTotalMilliCPU := int64(0)
+			podTotalMemory := int64(0)
+
+			for ix := range pod.Spec.Containers {
```

Comment:
For OpenShift, a cluster admin is the only user out of the box that can get
nodes, and they can get all pods.  I suspect we will have customers that
want to separate ability to get all pods from get all nodes, but its not a
major near term concern to hold up this PR.  Our policy allows that
configuration, I made a note to submit a follow-up to ensure its handled
gracefully.  In general, any of our describers that fetch related resources
need to handle it gracefully - especially when we get a resource in a
namespace, and fetch related non-namespaced items.

On Thu, Jul 30, 2015 at 11:48 AM, Brian Grant <notifications@github.com>
wrote:

> In pkg/kubectl/describe.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11788#discussion_r35885568>
> :
>
> >  		for _, pod := range pods {
> > -			fmt.Fprintf(out, "  %s\t%s\n", pod.Namespace, pod.Name)
> > +			podTotalMilliCPU := int64(0)
> > +			podTotalMemory := int64(0)
> > +
> > +			for ix := range pod.Spec.Containers {
>
> @derekwaynecarr <https://github.com/derekwaynecarr> Do we expect this to
> work for clusters with security policy enabled? An arbitrary client might
> not be able to read all the pods.
>
> That issue doesn't have to block this PR, since the existing code has the
> same problem, but it would be nice if this at least gracefully handled that
> case.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11788/files#r35885568>
> .
>